lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -199,25 +201,26 @@ data TransactionLayer k tx = TransactionLayer
 
     , updateTx
         :: tx
-        -> ExtraTxBodyContent
+        -> TxUpdate
         -> Either ErrUpdateSealedTx tx
         -- ^ Update tx by adding additional inputs and outputs
     }
     deriving Generic
 
--- | Describes modifications that can be made to a `TxBody` using
--- `updateTx`.
-data ExtraTxBodyContent = ExtraTxBodyContent
+-- | Describes modifications that can be made to a `Tx` using `updateTx`.
+data TxUpdate = TxUpdate
     { extraInputs :: [TxIn]
     , extraCollateral :: [TxIn]
        -- ^ Only used in the Alonzo era and later. Will be silently ignored in
        -- previous eras.
     , extraOutputs :: [TxOut]
+    , newExUnits :: ScriptWitnessIndex -> ExecutionUnits -> ExecutionUnits
+       -- ^ Adjust execution units on existing redeemers.
     , newFee :: Coin -> Coin
```

Comment:
I thought I went with `modifyFee` here, which would make more sense when it's a `a -> a`. If you feel like renaming to say `modifyExUnits` / `modifyFee`, feel free.